### ISSUE TITLE
Not all songs have lyrics through the extension, even if Spotify otherwise has them

### ISSUE THREAD

**Issue by Shroom2020** [surfbryce/beautiful-lyrics#58]
It seems some music, even though it has [long-standing entries in Musixmatch](https://www.musixmatch.com/lyrics/Cults/Gilded-Lily-Sped-Up), will simply not have its lyrics detected by the extension's page. This happens with both the Marketplace and pure JS extension versions.  

![image](https://github.com/surfbryce/beautiful-lyrics/assets/68300730/bd97be3a-5148-491e-bfed-269ed8cc03f3)
![image](https://github.com/surfbryce/beautiful-lyrics/assets/68300730/d1890100-f856-408d-ae66-c961ad6c9d1d)

**Comment by surfbryce** [comment]
I think this might be an issue with Spotify or your network as it seems to get the lyrics fine on my end;

<img width="1722" alt="image" src="https://github.com/surfbryce/beautiful-lyrics/assets/80861823/8148e641-1c45-4c99-9fe3-45ad9cccf2cf">

Can you:
1. Run the command "spicetify apple enable-devtools"
2. Then click the three dots in the top left
3. Click develop
4. Click on Developer Tools
5. Click the console tab

and finally screenshot everything from the console - this will help me figure out what's wrong. Although - unfortunately - if you have completely closed Spotify by the time you read this since having ran into this issue there might not be anything of use/help for me. If you keep this mode on and run into any other song that has this issue though immediately follow the steps I've written out and post the screenshots here!

**Comment by Shroom2020** [comment]
Here's my DevTools log from just after launching Spicetify.  

![Screenshot 2023-07-11 at 17 47 59](https://github.com/surfbryce/beautiful-lyrics/assets/68300730/f8f8ce76-62bb-4d84-91c8-b7c3e1f2bacf)
![Screenshot 2023-07-11 at 17 48 36](https://github.com/surfbryce/beautiful-lyrics/assets/68300730/a6720818-b814-4d9a-9727-5acda0c92106)

**Comment by surfbryce** [comment]
@Shroom2020 Yeah unfortunately it has to be on a new song that also has this issue since the lyrics result is cached on your end and so it will always just return that.

However, what you can do is clear the cache.

If you go back to the Developer Tools and click on the Application tab, then expand the Cache Storage list item, and finally right click on the Beautiful Lyrics entry you can delete all the cached lyrics/track information.

<img width="1071" alt="image" src="https://github.com/surfbryce/beautiful-lyrics/assets/80861823/d6ea02ea-0090-4659-9aab-4eed1251e64c">

Once you do this if you go to the same song again it should (hopefully in this instance) fail to get the lyrics; if it fails again please go to the console and screenshot everything you can get from it. There should be some errors there from my extension

**Comment by Shroom2020** [comment]
![Screenshot 2023-07-11 at 20 11 22](https://github.com/surfbryce/beautiful-lyrics/assets/68300730/9627988c-b787-431a-a055-b5816935d11b)

Not that I see.

**Comment by surfbryce** [comment]
@Shroom2020 did you delete the Beautiful Lyrics cache? Before deleting the cache make sure you are on a different song, then delete the cache, close spotify, and then reopen Spotify and have the console open and ready. Then go to that song that is failing to load Lyrics for you.

**Comment by surfbryce** [comment]
@Shroom2020 Can you provide confirmation that you cleared the cache by deleting the BeautifulLyrics entry? I can't tell if I'm going in a conversation loop or not.

Can you also try checking out the Network tab and ticking the Fetch/XHR tab.

You should see at least one entry that looks like this;
<img width="194" alt="image" src="https://github.com/surfbryce/beautiful-lyrics/assets/80861823/0cadb4d6-8186-48df-9cd4-3f06a77474cd">

If you click on it, screenshots of the "Headers" and "Response" tabs would be awesome.

The main thing is that if my backend doesn't have lyrics and Spotify does have line-synced lyrics than it'll use Spotifys lyrics. However, Gilded Lily actually does have lyrics from my backend. So in either case it should be loading them for you.

Those Spotify lyrics are retrieved on your end as well - which is why I'm figuring it's an issue with your Spotify client on your network.

**Comment by Shroom2020** [comment]
@surfbryce Yes, I did clear the cache, restart Spotify, and everything. Here's what happened on the network side:  

![Screenshot 2023-07-11 at 21 01 17](https://github.com/surfbryce/beautiful-lyrics/assets/68300730/2609f33b-95aa-42a9-8dd1-9eb24947a706)
![Screenshot 2023-07-11 at 21 01 34](https://github.com/surfbryce/beautiful-lyrics/assets/68300730/48b0c2ad-1db0-40e5-8265-f25d084662ca)

If I open this endpoint in a browser, it also shows absolutely nothing. The request is successful but I'm not getting anything of use back, it seems.  

![Screenshot 2023-07-11 at 21 01 45](https://github.com/surfbryce/beautiful-lyrics/assets/68300730/a7e6b1d1-cbb9-4c76-a500-1908b0266557)

**Comment by surfbryce** [comment]
@Shroom2020 is this what appears for the Gilded Lily (Sped Up) song? I looked up the ISRC code you just shared (QM8QH2000855) and no song exists with that. Which should be impossible since every song is required to have an ISRC code when registered with Spotify/Apple Music.

The record code for Gilded Lily (Sped Up) is USQX92206421 and for the normal version its USQX92206420 and the slowed down version is USQX92206422.

If you clear the cache again and then load up the Gilded Lily song see if you see this at all. I don't think it's possible for us to see the requests Spotify is making unfortunately so I don't know whether or not the requests on that end are failing.

There's something very wrong happening on your end though - I just don't know what it is yet

**Comment by Shroom2020** [comment]
@surfbryce I can't really comment on that then. I have even less of a clue of what Spotify, and your extension, is trying to do here.

**Comment by surfbryce** [comment]
@Shroom2020 That's the one thing I'm not sure about then - because on my side Spotify is returning back to me the ISRC that matches up with the US country code. But for some reason it's returning QM (alternative US country-code) for you.

This is definitely an issue with Spotify if it's returning different ISRC code for the same track since I am using their API to get the track information (and in turn their ISRC code). I'll have to look and see if there are any other codes returned for that song which may help me get the right ISRC.

Is this the only song you've come across that has this issue btw? From the title you made it seem like it happens for other songs as well but this is the only one you presented so far. It'll be helpful if you are aware of any others that do this for you.

However, there is something I can do to fix this issue so at least it's able to get the Spotify line-synced lyrics for now. I'll just forcefully add the current song id until the lyrics-fetcher for the Spotify portion since it should always at the least be getting the Spotify lyrics for the current version of the song (obviously)

**Comment by Shroom2020** [comment]
@surfbryce I have noticed it happen with other music, but I haven't kept track of exactly what it was. This is the only one I remember because I was looking forward to seeing how it looks with BeautifulLyrics.

**Comment by surfbryce** [comment]
Oh one more thing @Shroom2020 - what country are you using this from and do you have a VPN on per-chance? If you do what VPN is that country from?

I did a little research and it's possible that based on your country there may be a different registered variant of the song which causes it to have a different ISRC code... which is just weird but possible!

**Comment by Shroom2020** [comment]
@surfbryce I'm from the UK. No VPN here.

**Comment by MrGenia** [comment]
I'm experiencing the same problem with the song "My Mother Told Me" by "The Hound + The Fox" (ISRC QZGWX2190998). As suggested, I have cleared all cache. I'm using Spotify in Ukraine without a VPN. Also, the Reverse Lookup feature in ISRC Finder works fine for that song. What I did was delete all Spicetify extensions, then uninstall Spicetify and Spotify, reinstall Spotify and Spicetify with only this extension, and faced the same problem. But as a bonus, I now have the "Now Playing View" function added, haha. Another thing I should mention is that I'm unable to scroll through the lyrics on the original Spotify lyrics tab
![1](https://github.com/surfbryce/beautiful-lyrics/assets/48743210/90df5b87-4b17-4166-aedd-98f18ae7f79d)
![2](https://github.com/surfbryce/beautiful-lyrics/assets/48743210/43dfad07-371e-447b-a103-a530d4a5f2e9)
![3](https://github.com/surfbryce/beautiful-lyrics/assets/48743210/7a3ba1f4-05b5-4a22-b8bc-fe8bcf8db266)
![4](https://github.com/surfbryce/beautiful-lyrics/assets/48743210/11d50b12-e97e-4445-b2bc-54cca59ce246)
![5](https://github.com/surfbryce/beautiful-lyrics/assets/48743210/4af492d1-97bc-4ef2-96f3-ef037ddf0a2e)
![6](https://github.com/surfbryce/beautiful-lyrics/assets/48743210/9b26d858-5243-4735-aba9-46711457e725)

**Comment by surfbryce** [comment]
@MrGenia Actually in this case the extension is working as intended - I cannot trust static lyrics from Spotify because Spotify will often times show incorrect lyrics or lyrics from another song when it comes to text-based lyrics.

Although, even when using Spotify to view the original lyrics they aren't present for me so it seems like those are regionally licensed lyrics as well.

**Comment by Shroom2020** [comment]
@surfbryce So how'll the resolution on this work?

**Comment by surfbryce** [comment]
@Shroom2020

In short, right now I get all the possible track-ids that have the current tracks ISRC; however, because the right ISRC isn't being returned for you - it's not including the current track-id being played - so the solution is to just forcefully add the current track-id so that it always gets the lyrics for the playing track.

This solution doesn't interfere with anything or mess with the priorities; it simply forces the current track-id to be in our spotify track selection if it doesn't exist yet so that we can always just fallback onto the Spotify lyrics (if they aren't static).

I've been extremely busy recently so I haven't been able to push out an update for Beautiful Lyrics yet; but, I already have this working and implemented on my end. I have to finish some other suggestions before I push out the update so please bear with me on this!

**Comment by surfbryce** [comment]
@Shroom2020 This fix should be published - sorry for not getting this out sooner. I was primarily waiting to publish once I've finished the other features I wanted to get to.

Let me know if this is still happening. I've updated the cache versions so that new lyrics and track information will be retrieved. Hopefully, the track-information helps out with the ISRC issue and the lyrics should let that new forced spotify track inclusion to work out and fix the issue you were encountering.

Will close this issue once it's been confirmed that it works.
